File: algorithms/1252.cells-with-odd-values-in-a-matrix/solution.rs
impl Solution {
    pub fn odd_cells(n: i32, m: i32, indices: Vec<Vec<i32>>) -> i32 {
        let mut matrix: Vec<Vec<i32>> = vec![vec![0; m as usize]; n as usize];
        let mut count: i32 = 0;

        // apply the increment to all indices
        for ind in indices {
            let (ri, ci) = (ind[0] as usize, ind[1] as usize);

            // row
            for cell in matrix[ri].iter_mut() {
               *cell += 1;
            }
            // column
            for row in matrix.iter_mut() {
                row[ci] += 1;
            }
        }

        for row in matrix {
            count += row.iter().filter(|cell| *cell % 2 != 0).count() as i32;
        }

        count
    }
}

File: algorithms/1252.cells-with-odd-values-in-a-matrix/solution.rb
def odd_cells(n, m, indices)
  matrix = Array.new(n) { Array.new(m, 0) }
  count = 0

  # apply the increment to all indices
  indices.each do |ind|
    ri, ci = ind[0], ind[1]
    # row
    matrix[ri].map! { |cell| cell = cell + 1 }
    # column
    matrix.each { |row| row[ci] = row[ci] + 1 }
  end

  # matrix.flatten.count {|cell| cell % 2 != 0 }
  matrix.each do |row|
    count += row.count { |cell| cell % 2 != 0 }
  end

  count
end
